nixos/tests/seafile.nix

```
@@ -0,0 +1,120 @@
+import ./make-test-python.nix ({ pkgs, ... }:
+  let
+    client = { config, pkgs, ... }: {
+      virtualisation.memorySize = 256;
+      environment.systemPackages = [ pkgs.seafile-shared ];
+    };
+  in {
+    name = "seafile";
+    meta = with pkgs.stdenv.lib.maintainers; {
+      maintainers = [ kampfschlaefer ];
+    };
+
+    nodes = {
+      server = { config, pkgs, ... }: {
+        virtualisation.memorySize = 512;
+        services.seafile = {
+          enable = true;
+          ccnetSettings.General.SERVICE_URL = "http://server";
+          adminEmail = "admin@example.com";
+          initialAdminPassword = "seafile_password";
+        };
+        services.nginx = {
+          enable = true;
+          virtualHosts."server" = {
+            locations."/".proxyPass = "http://unix:/run/seahub/gunicorn.sock";
+            locations."/seafhttp" = {
+              proxyPass = "http://127.0.0.1:8082";
+              extraConfig = ''
+                rewrite ^/seafhttp(.*)$ $1 break;
+                client_max_body_size 0;
+                proxy_connect_timeout  36000s;
+                proxy_read_timeout  36000s;
+                proxy_send_timeout  36000s;
+                send_timeout  36000s;
+                proxy_http_version 1.1;
+              '';
+            };
+          };
+        };
+        networking.firewall = { allowedTCPPorts = [ 80 ]; };
+      };
+      client1 = client pkgs;
+      client2 = client pkgs;
+    };
+
+    testScript = ''
+      start_all()
+
+      with subtest("start seaf-server"):
+          server.wait_for_unit("seaf-server.service")
+          server.wait_for_file("/run/seafile/seafile.sock")
+
+      with subtest("start seahub"):
+          server.wait_for_unit("seahub.service")
+          server.wait_for_unit("nginx.service")
+          server.wait_for_file("/run/seahub/gunicorn.sock")
+
+      with subtest("client1 connect"):
```

small seahub smoke test, part 2:
check whether the Log In page of SeaHub can be retrieved.

```suggestion
      with subtest("client1 fetch seahub page"):
          client1.succeed("curl -L http://server | grep 'Log In' >&2")
      
      with subtest("client1 connect"):
```